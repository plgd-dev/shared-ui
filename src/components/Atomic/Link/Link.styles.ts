import { css } from '@emotion/react'
import { getTheme, getThemeColor, ThemeType } from '../_theme'
import { fontPrimary } from '../_utils/commonStyles'

export const link = (theme: ThemeType) => css`
    font-family: ${getTheme(theme, `Global.fontPrimary`, fontPrimary)};
    text-decoration-line: underline;
    color: ${getThemeColor(theme, `Link.normal.color`)};
    font-weight: 400;
    font-size: 12px;
    line-height: 150%;

    &:hover {
        text-decoration: none;
    }
`

export const big = (theme: ThemeType) => css`
    color: ${getThemeColor(theme, `Link.big.color`)};
    border-bottom: 2px solid ${getThemeColor(theme, `Link.big.color`)};
    text-decoration: none !important;
    transition: all 0.25s;
    padding-bottom: 2px;
    font-weight: bold;

    &:hover {
        text-decoration: none;
        border-bottom-color: transparent;
    }
`
