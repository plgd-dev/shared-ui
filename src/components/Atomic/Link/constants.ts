export const linkSizes = {
    BIG: 'big',
    NORMAL: 'normal',
} as const
