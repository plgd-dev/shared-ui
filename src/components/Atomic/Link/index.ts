export { default } from './Link'
export * from './Link'

export * from './constants'
