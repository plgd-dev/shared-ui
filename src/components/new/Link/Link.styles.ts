import { css } from '@emotion/react'

export const link = css`
    font-family: 'Poppins', sans-serif;
    text-decoration-line: underline;
    color: #81868c;
    font-weight: 400;
    font-size: 12px;
    line-height: 150%;

    &:hover {
        text-decoration: none;
    }
`
