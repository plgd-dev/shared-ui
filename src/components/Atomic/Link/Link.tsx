import { FC, MouseEvent } from 'react'
import { defaultProps, Props } from './Link.types'
import * as styles from './Link.styles'

const Link: FC<Props> = (props) => {
    const { children, dataTestId, disabled, href, onClick, preventDefault, target } = { ...defaultProps, ...props }

    const handleOnClick = (e: MouseEvent<HTMLElement>) => {
        if (!disabled && onClick) {
            preventDefault && e.preventDefault()
            onClick(e)
        }
    }

    return (
        <a css={styles.link} href={href || '#'} onClick={handleOnClick} target={target} data-test-id={dataTestId}>
            {children}
        </a>
    )
}

Link.displayName = 'Link'
Link.defaultProps = defaultProps

export default Link
