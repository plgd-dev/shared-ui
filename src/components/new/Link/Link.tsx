import { FC, MouseEvent } from 'react'
import { Props } from './Link.types'
import * as styles from './Link.styles'

const Link: FC<Props> = (props) => {
    const { children, disabled, href, onClick, preventDefault, target } = props

    const handleOnClick = (e: MouseEvent<HTMLElement>) => {
        if (!disabled && onClick) {
            preventDefault && e.preventDefault()
            onClick(e)
        }
    }

    return (
        <a css={styles.link} href={href || '#'} onClick={handleOnClick} target={target}>
            {children}
        </a>
    )
}

Link.displayName = 'Link'

export default Link
