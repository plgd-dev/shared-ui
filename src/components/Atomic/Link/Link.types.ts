import { HTMLAttributeAnchorTarget, MouseEvent, ReactNode } from 'react'

import { linkSizes } from './constants'
export type LinkSizesType = (typeof linkSizes)[keyof typeof linkSizes]

export type Props = {
    children: ReactNode
    dataTestId?: string
    disabled?: boolean
    href?: string
    onClick?: (e: MouseEvent<HTMLElement>) => void
    preventDefault?: boolean
    size?: LinkSizesType
    target?: HTMLAttributeAnchorTarget
    tabIndex?: number
}

export const defaultProps = {
    size: linkSizes.NORMAL,
    tabIndex: 1,
}
