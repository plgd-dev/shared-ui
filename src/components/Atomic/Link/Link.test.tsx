import React from 'react'
import { fireEvent, render } from '@testing-library/react'
import Link from './Link'
import { forEach } from 'lodash'

describe('<Headline>', () => {
    it('render correctly - snapshot', () => {
        const { asFragment } = render(
            <div>
                <Link>Link no href</Link>
                <Link href='//www.plgd.dev'>Link with href</Link>
            </div>
        )

        expect(asFragment()).toMatchSnapshot()
    })

    it('onCLick', async () => {
        const callback1 = jest.fn()
        const callback2 = jest.fn()

        const { container } = render(
            <div>
                <Link dataTestId='link-test' href='//www.plgd.dev' onClick={callback1}>
                    Link with href
                </Link>
                <Link dataTestId='link-test' disabled={true} href='//www.plgd.dev' onClick={callback2}>
                    disabled
                </Link>
            </div>
        )

        const links = container.querySelectorAll('[data-test-id="link-test"]')
        forEach(links, (button) => {
            fireEvent.click(button)
        })

        expect(callback1).toHaveBeenCalled()
        expect(callback2).not.toHaveBeenCalled()
    })
})
