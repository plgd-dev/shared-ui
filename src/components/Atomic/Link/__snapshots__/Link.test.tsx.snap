// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Headline> render correctly - snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: 'Poppins',sans-serif;
  text-decoration-line: underline;
  color: #81868C;
  font-weight: 400;
  font-size: 12px;
  line-height: 150%;
}

.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div>
    <a
      class="emotion-0"
      href="#"
    >
      Link no href
    </a>
    <a
      class="emotion-0"
      href="//www.plgd.dev"
    >
      Link with href
    </a>
  </div>
</DocumentFragment>
`;
